### ISSUE TITLE
"test_batch0_gt.jpg" is not seen after training

### ISSUE THREAD

**Issue by lipond** [ultralytics/yolov5#1361]
I run the training using yolov5 and when epoch0 is done I got "test_batch0_pred.jpg" and "test_batch0_labels.jpg". Did not see "test_batch0_gt.jpg" documented in https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data. Is it a typo or bug? Thanks.

**Comment by glenn-jocher** [comment]
@lipond ah yes you are correct! We did a recent PR https://github.com/ultralytics/yolov5/pull/1337 to replace 'gt' with 'labels' throughout the code, but we forgot to update the tutorial with the change. Thanks for spotting!

**Comment by glenn-jocher** [comment]
Ok, I've updated the tutorial now with this correction! Closing issue.

**Comment by lipond** [comment]
Hi @glenn-jocher , I run training in my custom dataset and found the "test_batch0_labels.jpg" do not display the right groundtruth, just as following:
![image](https://user-images.githubusercontent.com/7008933/99223954-68f6c780-2820-11eb-9770-67e55808e9ec.png)
And the test results seemed not right. So I doubt if the testloader has a bug?

I run the command "python train.py --img 640 --batch 48 --epochs 140 --data person-wide-1c.yaml --weights yolov5s.pt". If I don't use yolov5 right, please let me know and thank you.

**Comment by glenn-jocher** [comment]
@lipond the repo operates correctly, the source of error is the labels you have supplied. See https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

**Comment by lipond** [comment]
@glenn-jocher I carefully checked my data and labels and can make sure that they are right and one to one correspondences. Could you please help me see the dataset? Thanks.

**Comment by glenn-jocher** [comment]
@lipond you are seeing the dataset in the image you uploaded. Your labels are not correct.

**Comment by lipond** [comment]
@glenn-jocher Yes, that is what I am confused with. The labels of dataset I used are correct and I do not know why they are not displayed correctly.

**Comment by glenn-jocher** [comment]
@lipond they are displayed exactly the way they should be displayed. Your underlying data is being displayed correctly. You draw the conclusion.

**Comment by lipond** [comment]
@glenn-jocher I think I am sure that the labels I provided are not like the way they are displayed. Firstly the box numbers are not correct, let alone the box locations. Let me give you some examples.

The left bottom image in the "test_batch0_labels.jpg" is "kitti_kitti_train_006825.jpg", and there are 3 boxes displayed(what you can see from that image). But its label I provide is like this(only 2 boxes):
0	0.364371980676	0.539541547278	0.0182769726248	0.0916905444128
0	0.34847020934	0.53553008596	0.0202898550725	0.096848137536

The left top image in the "test_batch0_labels.jpg" is "coco_000000290623.jpg", and there is only 1 box displayed there. But its label I provied is like this (2 boxes):
0 0.67796875 0.348011363636 0.1640625 0.4125
0 0.31234375 0.222869318182 0.06 0.216193181818

**Comment by lipond** [comment]
But what confused me more is botth "test_batch0_pred.jpg" and "train_batch*.jpg" seem correct. It indicates the training routine and label loading are both normal. So I wonder if the labels loading in the test routine is with bug. I hope I am wrong and just I use yolov5 by a wrong way.

**Comment by glenn-jocher** [comment]
@lipond if you believe there is a bug then please raise a new bug report issue, and most importantly please provide a reproducible example, i.e. a colab notebook that reproduces the error. Here are the bug report details:

Please note that most technical problems are due to:

- **Your modified or out-of-date code**. If your issue is not reproducible in a new `git clone` version of this repo we can not debug it. Before going further run this code and verify your issue persists:
```bash
$ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest
$ cd yolov5_new
$ python detect.py  # verify detection

# CODE TO REPRODUCE YOUR ISSUE HERE
```

- **Your custom data.** If your issue is not reproducible in one of our 3 common datasets ([COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [COCO128](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), or [VOC](https://github.com/ultralytics/yolov5/blob/master/data/voc.yaml)) we can not debug it. Visit our [Custom Training Tutorial](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) for guidelines on training your custom data. Examine `train_batch0.jpg` and `test_batch0.jpg` for a sanity check of your labels and images.

- **Your environment.** If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new [venv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/), and then install requirements.

If none of these apply to you, we suggest you close this issue and raise a new one using the **Bug Report template**, providing screenshots and **minimum viable code to reproduce your issue**. Thank you!


## Requirements

Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.6`. To install run:
```bash
$ pip install -r requirements.txt
```


## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab Notebook** with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- **Kaggle Notebook** with free GPU: [https://www.kaggle.com/models/ultralytics/yolov5](https://www.kaggle.com/models/ultralytics/yolov5)
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) 
- **Docker Image** https://hub.docker.com/r/ultralytics/yolov5. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) ![Docker Pulls](https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker)

## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.

**Comment by lipond** [comment]
@glenn-jocher Sorry, I found out it is because I modified the labels wrongly in the process of training. Now it is running normally. Thanks anyway!
